src/main/java/com/bigsonata/swarm/interop/Message.java

```diff
@@ -76,9 +40,63 @@ public Message(byte[] bytes) throws IOException {
     unpacker.close();
   }
 
-  public Message(String type, Map data, String nodeID) {
+  public static Map<String, Object> unpackMap(MessageUnpacker unpacker) throws IOException {
+    int mapSize = unpacker.unpackMapHeader();
+    Map<String, Object> result = new HashMap<>(6);
+    while (mapSize > 0) {
+      String key = null;
+      // unpack key
+      if (unpacker.getNextFormat() == MessageFormat.NIL) {
+        unpacker.unpackNil();
+      } else {
+        key = unpacker.unpackString();
+      }
+      // unpack value
+      MessageFormat messageFormat = unpacker.getNextFormat();
+      Object value;
+
+      switch (messageFormat.getValueType()) {
+        case BOOLEAN:
+          value = unpacker.unpackBoolean();
+          break;
+        case FLOAT:
+          value = unpacker.unpackFloat();
+          break;
+        case INTEGER:
+          value = unpacker.unpackInt();
+          break;
+        case NIL:
+          value = null;
+          unpacker.unpackNil();
+          break;
+        case STRING:
+          value = unpacker.unpackString();
+          break;
+        case MAP:
+          value = unpackMap(unpacker);
```

Comment:
Recursion!

Reply:
@jayudey-wf WDYT?

Reply:
i 👀 how `locust4j` solved this issue and one of the places that it looked like it changed was this approach so i thought to update it in line w/ their approach